src/Jackalope/Version/VersionHandler.php

```diff
@@ -103,14 +107,20 @@ public function checkinItem($path)
         }
 
         if ($node->isModified()) {
-            throw new InvalidItemStateException(sprintf(
-                'Node "%s" contains unsaved changes',
-                $path
-            ));
+            throw new InvalidItemStateException(
+                sprintf(
+                    'Node "%s" contains unsaved changes',
+                    $path
+                )
+            );
         }
 
+        /** @var NodeInterface $baseVersionNode */
+        $baseVersionUuid = $node->getProperty('jcr:baseVersion')->getString();
+        $baseVersionNode = $this->objectManager->getNodeByIdentifier($baseVersionUuid, 'Version\Version');
```

Comment:
@dantleech for 2.0 it would be really nice if the transport could discover that this is a version instead of we having to tell it explicitly. then these two lines could read `$baseVersionNode = $node->getPropertyValue('jcr:baseVersion');`